File: Seccion 18 - Clases en C++/Seccion_18_Dart/03_ejercicio_03.dart

import 'dart:io';

class Tablero {

  int x, y;

  Tablero(this.x, this.y);

  void moverArriba(int n) {
    y += n;
  }

  void moverAbajo(int n) {
    y -= n;
  }

  void moverDerecha(int n) {
    x += n;
  }

  void moverIzquierda(int n) {
    x -= n;
  }

  int getX() {
    return x;
  }

  int getY() {
    return y;
  }
}

void main() {

  Tablero obj1;
  int opcion, n, x, y;

  print("Digite la posición inicial del objeto: ");
  stdout.write("Posición X: ");
  x = int.parse(stdin.readLineSync());
  stdout.write("Posición Y: ");
  y = int.parse(stdin.readLineSync());

  obj1 = new Tablero(x, y);

  do {
    print("\n\t.:MENU:.");
    print("1. Mover Arriba.");
    print("2. Mover Abajo.");
    print("3. Mover Derecha.");
    print("4. Mover Izquierda.");
    print("5. Salir.");
    stdout.write(("Opción: "));
    opcion = int.parse(stdin.readLineSync());

    if (opcion > 0 && opcion <= 4) {
      stdout.write("Digite las posiciones para mover: ");
      n = int.parse(stdin.readLineSync());
    }

    switch (opcion) {
      case 1:
        obj1.moverArriba(n);
        break;
      
      case 2:
        obj1.moverAbajo(n);
        break;
      
      case 3:
        obj1.moverDerecha(n);
        break;
      
      case 4:
        obj1.moverIzquierda(n);
        break;
      
      case 5:
        break;
      
      default:
        print("Opción inválida.");
        break;
    }

    print("Posición actual del objeto -> (X, Y) : (${obj1.getX()}, ${obj1.getY()})");
  } while (opcion != 5);
}

File: Seccion 18 - Clases en C++/Seccion_18_Python/03_ejercicio_03.py

class Tablero:
    
    def __init__(self, x: int, y: int):
        self.x = x
        self.y = y
    
    def mover_arriba(self, n: int):
        self.y += n
    
    def mover_abajo(self, n: int):
        self.y -= n
    
    def mover_derecha(self, n: int):
        self.x += n
    
    def mover_izquierda(self, n: int):
        self.x -= n
    
    def get_X(self):
        return self.x

    def get_Y(self):
        return self.y

def main():
    
    opcion = 0
    
    print("Digite la posición inicial del objeto:")
    x = int(input("Posición X: "))
    y = int(input("Posición Y: "))
    
    ob1 = Tablero(x, y)
    
    while opcion != 5:
        print("\n\t.:MENU:.")
        print("1. Mover Arriba.")
        print("2. Mover Abajo.")
        print("3. Mover Derecha.")
        print("4. Mover Izquierda.")
        print("5. Salir.")
        opcion = int(input("Opción: "))
        
        if opcion > 0 and opcion <= 4:
            n = int(input("Digite las posiciones a mover: "))
        
        if opcion == 1:
            ob1.mover_arriba(n)
        elif opcion == 2:
            ob1.mover_abajo(n)
        elif opcion == 3:
            ob1.mover_derecha(n)
        elif opcion == 4:
            ob1.mover_izquierda(n)
        elif opcion == 5:
            pass
        else:
            print("Opción inválida.")
            continue
    
        print(f"Posición actual del objeto -> (X, Y): ({ob1.get_X()}, {ob1.get_Y()})")

if __name__ == "__main__":
    main()